### PR TITLE
[SPARK-29452][WEBUI] Improve Storage tab tooltip

### PR DESCRIPTION
**Pull request by iRakson**
### What changes were proposed in this pull request?
Added Tootips for each column in storage tab of Web UI.

### Why are the changes needed?
Tooltips will help users in understanding columns of storage tabs.

### Does this PR introduce any user-facing change?
Yes

### How was this patch tested?
Manually Tested.

### CODE REVIEW

**Comment by wangyum** [comment]
ok to test

**Comment by uncleGen** [comment]
retest this please.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Hi, @iRakson .
Could you fix the UT `StoragePageSuite.rddTable`?
- https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112544/testReport/org.apache.spark.ui.storage/StoragePageSuite/rddTable/

**Comment by iRakson** [comment]
@dongjoon-hyun yes working on this issue.

File: core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      "Replication factor of the RDD"
+
+  val CACHED_PARTITIONS =
+    "Number of partitions cached"
+
+  val FRACTION_CACHED =
+    "Fraction of RDD cached"
+
+  val SIZE_IN_MEMORY =
+    "Total size in the memory"
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala]: the memory -> memory
Total size of what? (cached partitions)

File: core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  val STORAGE_LEVEL =
+    "StorageLevel displays where RDD is stored, " +
+      "Format of RDD (Serialized or De-serialized) and" +
+      "Replication factor of the RDD"
+
+  val CACHED_PARTITIONS =
+    "Number of partitions cached"
+
+  val FRACTION_CACHED =
+    "Fraction of RDD cached"
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala]: Interesting, I actually dont' know whether is the fraction of partitions or data cached. That's not quite the same. Would be good to figure it out and clarify here

File: core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+private[ui] object ToolTips {
+  val ID =
+    "ID of RDD"
+
+  val RDD_NAME =
+    "Name of persisted RDD"
+
+  val STORAGE_LEVEL =
+    "StorageLevel displays where RDD is stored, " +
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala]: Format, Replication, Serialized, De-serialized should not be capitalized
RDD -> the RDD

File: core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+package org.apache.spark.ui.storage
+
+private[ui] object ToolTips {
+  val ID =
+    "ID of RDD"
+
+  val RDD_NAME =
+    "Name of persisted RDD"
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala]: Use 'persisted' or 'cached' consistently

File: core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ui.storage
+
+private[ui] object ToolTips {
+  val ID =
+    "ID of RDD"
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/storage/ToolTips.scala]: This doesn't add anything

**Review by srowen — CHANGES_REQUESTED** [review]

**Review by srowen — APPROVED** [review]

**Comment by iRakson** [comment]
cc @wangyum

**Comment by srowen** [comment]
Merged to master

**Comment by srowen** [comment]
Merged to master. @iRakson what's your JIRA ID? I can assign you the issue.

**Comment by iRakson** [comment]
@srowen Rakesh Raushan

**Pull request closed without merging**